### PR TITLE
Manual upgrade to float the deps since automation is failing

### PR DESCRIPTION
**Pull request by vaikas**
# Changes

- Manually upgraded the deps so they are not locked. Then ran:
```
./hack/update-deps.sh --upgrade --release v0.21
./hack/update-codegen.sh
```

**NOTE** For some reason kafka now dropped to 0.19 but at least before this change, when I tried to run the above, it failed with:

```
vaikas@villes-mbp eventing-autoscaler-keda % ./hack/update-deps.sh --upgrade --release v0.21
=== Update Deps for Golang
--- Upgrading to v0.21
Using 'GOPROXY=direct'.
Floating deps to knative.dev/eventing@master knative.dev/eventing-awssqs@master knative.dev/eventing-kafka@master knative.dev/eventing-redis@master knative.dev/hack@master knative.dev/pkg@master
verifying knative.dev/eventing-awssqs@v0.20.0/go.mod: checksum mismatch
	downloaded: h1:6d0Y+8MPDqiSJtp2X0w5PAlrqu1kOjfgomTZbbwr+Xc=
	go.sum:     h1:RYjhn7/5NIfdUOtgLLSHFVO2kQ6v0zIb+fIcWupU7S8=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
vaikas@villes-mbp eventing-autoscaler-keda % git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

My thinking is that we at least get the automation to not fail consistently, then tackle the kafka dep as a separate issue? Thoughts @zroubalik @lionelvillard 

But let's also wait until we see what happens on the last automation run here wrt. this repo:
https://github.com/knative-sandbox/knobots/runs/1784255342?check_suite_focus=true

-
-

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Comment by vaikas** [comment]
An update, automation is still failing:

```
Run # Determine the name of the go module.
=== Update Deps for Golang
Upgrading to v0.21
  Using 'GOPROXY=direct'.
  Floating deps to knative.dev/eventing@master knative.dev/eventing-awssqs@master knative.dev/eventing-kafka@master knative.dev/eventing-redis@master knative.dev/hack@master knative.dev/pkg@master
  verifying knative.dev/eventing-awssqs@v0.20.0/go.mod: checksum mismatch
  	downloaded: h1:6d0Y+8MPDqiSJtp2X0w5PAlrqu1kOjfgomTZbbwr+Xc=
  	go.sum:     h1:RYjhn7/5NIfdUOtgLLSHFVO2kQ6v0zIb+fIcWupU7S8=
  
  SECURITY ERROR
  This download does NOT match an earlier download recorded in go.sum.
  The bits may have been replaced on the origin server, or an attacker may
  have intercepted the download attempt.
  
  For more information, see 'go help module-auth'.
Error: Process completed with exit code 1.
```

File: go.mod [hunk 1]
```diff
@@ -10,12 +10,12 @@ require (
 	k8s.io/apimachinery v0.20.2
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
 	k8s.io/code-generator v0.20.0
-	knative.dev/eventing v0.20.1
-	knative.dev/eventing-awssqs v0.20.0
-	knative.dev/eventing-kafka v0.20.0
-	knative.dev/eventing-redis v0.20.0
+	knative.dev/eventing v0.20.1-0.20210128132430-1725902f7e39
+	knative.dev/eventing-awssqs v0.20.1-0.20210128095331-4c73bf15f7c8
+	knative.dev/eventing-kafka v0.19.1-0.20210128073630-1d9c884c16c6
```
Comment (slinkydeveloper) [inline go.mod:15]: Why eventing-kafka 0.19?

Comment (vaikas) [reply go.mod:15]: Great question @slinkydeveloper . For some reason or another that's what the upgrade pulls in.
But, I was thinking that we would at least fix the automation and upstream failures (which I think have been b0rk3d for weeks by now?) and then tackle kafka as a follow up?
I can create an issue to track that and link it in here?

Comment (zroubalik) [reply go.mod:15]: Agree

Comment (lionelvillard) [reply go.mod:15]: More precisely, why 0.19.1 when 0.19.3 exists? Also `20210128` seems to be a date (today), but which one? 

Anyway there is no API changes between 0.19 and 0.20 so all good.

**Pull request merged**